.cloudbuild/scripts/internal/artifacts/artifacts.go

```diff
@@ -0,0 +1,93 @@
+package artifacts
+
+import (
+	"context"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"cloud.google.com/go/storage"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+)
+
+// FindAndUpload finds all of the files referenced by the supplied patterns
+// and uploads them to the supplied GCS bucket. The supplied patterns are
+// expected to be fully-qualified paths, and will be searched without changing
+// the current directory.
+//
+// Artifacts from various paths will be aggregated into one place in the
+// bucket with the supplied prefix, using the file's base name to disambiguate,
+// so be wary of including multiple artifacts with teh same name.
+func FindAndUpload(ctx context.Context, bucketName, objectPrefix string, artifactPatterns []string) error {
+	log.Printf("Scanning for artifacts...")
+	artifacts := []string{}
+	for _, pattern := range artifactPatterns {
+		matches, err := filepath.Glob(pattern)
+		if err != nil {
+			log.Printf("Failed scanning for artifacts matching %q: %s", pattern, err.Error())
+			continue
+		}
+		artifacts = append(artifacts, matches...)
+	}
+
+	if len(artifacts) == 0 {
+		return nil
+	}
+
+	return Upload(ctx, bucketName, objectPrefix, artifacts...)
+}
+
+// Upload uploads a set of files to the indicated artefact bucket with the
+// supplied prefix.
+//
+// Note that artifacts from various paths will be aggregated into one place in
+// the bucket under the supplied prefix, using the file's base name to
+// disambiguate. Be wary of including multiple artifacts with the same name, as
+// later object may clobber earlier ones.
```

Comment:
Mention that errors are swallowed?

Or, better yet, upload as much as we can but return an aggregated error if we had any?

Reply:
Done.